src/nomad_simulations/schema_packages/properties/energies.py

```diff
@@ -20,33 +20,65 @@
 
 import numpy as np
 
-from nomad.metainfo import Quantity
+from nomad.metainfo import Quantity, Section, Context, SubSection
 
 if TYPE_CHECKING:
     from nomad.metainfo import Section, Context
     from nomad.datamodel.datamodel import EntryArchive
     from structlog.stdlib import BoundLogger
 
-from nomad_simulations.schema_packages.physical_property import PhysicalProperty
+from nomad_simulations.schema_packages.physical_property import (
+    PhysicalProperty,
+    PropertyContribution,
+)
 
+##################
+# Abstract classes
+##################
 
-class FermiLevel(PhysicalProperty):
+
+class BaseEnergy(PhysicalProperty):
     """
-    Energy required to add or extract a charge from a material at zero temperature. It can be also defined as the chemical potential at zero temperature.
+    Abstract physical property section describing some energy of a (sub)system.
     """
 
-    # ! implement `iri` and `rank` as part of `m_def = Section()`
-
-    iri = 'http://fairmat-nfdi.eu/taxonomy/FermiLevel'
-
     value = Quantity(
         type=np.float64,
         unit='joule',
         description="""
-        The value of the Fermi level.
+        The value of the energy.
         """,
     )
 
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+
+
+class EnergyContribution(BaseEnergy, PropertyContribution):
+    """
+    Abstract physical property section linking a property contribution to a contribution
+    from some method.
+    """
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+
+
+####################################
+# List of specific energy properties
+####################################
+
+
+class FermiLevel(BaseEnergy):
+    """
+    Physical property section describing the Fermi level, i.e., the energy required to add or extract a charge from a material at zero temperature.
```

Comment:
Please, do not change the description. This is copy-pasted from the taxonomy owl.